topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md

```diff
@@ -0,0 +1,321 @@
+
+If we give Snippy an annotated reference, it will silently run a tool called SnpEff which will figure out the effect of any changes on the genes and other features. If we just give Snippy the reference sequence alone without the annotations, it will not run SnpEff.
+
+We have an annotated reference built from the inferred *M. tuberculosis* [ancestral reference genome](https://zenodo.org/record/3497110) and the
+gene annotation from the [H37Rv strain](https://www.ncbi.nlm.nih.gov/nuccore/NC_000962.3) so will use it in this case.
+
+> ### {% icon hands_on %} Hands-on: Run Snippy
+>
+> 1. **Snippy** {% icon tool %} with the following parameters
+>   - *"Will you select a reference genome from your history or use a built-in index?"*: `Use a genome from history and build index`
+>   - *"Use the following dataset as the reference sequence"*: `Mycobacterium_tuberculosis_ancestral_reference.gbk`
+>   - *"Single or Paired-end reads"*: `Paired`
+>       - *"Select first set of reads"*: `Trimmomatic on https://zenodo.org/record/3531703/files/ERR550641_1.fastq.gz (R1 paired)`
+>       - *"Select second set of reads"*: `Trimmomatic on https://zenodo.org/record/3531703/files/ERR550641_2.fastq.gz (R2 paired)`
+>   - Under *"Advanced parameters"*
+>       - *"Minimum proportion for variant evidence"*: `0.1` (This is so we can see possible rare variants in our sample)
+>   - Under *"Output selection"* select the following:
+>       - *"The final annotated variants in VCF format"*
+>       - *"The alignments in BAM format"*
+>       - Deselect any others.
+>
+> 2. Inspect the Snippy VCF output
+>
+>    > ### {% icon question %} Questions
+>    >
+>    > 1. What type of variant is the first one in the list?
+>    >
+>    > 2. What was the effect of this variant on the coding region it was found in?
+>    >
+>    > > ### {% icon solution %} Solution
+>    > >
+>    > > 1. Substitution of a `C` to a `T`
+>    > >
+>    > > 2. According to SnpEff, it's a Synonymous change.
+>    > >
+>    > {: .solution}
+>    {: .question}
+{: .hands_on}
+
+**RECAP**: So far we have taken our sample reads, cleaned them up a bit, checked for taxonomic assocation, compared the reads with our reference sequence and then called variants (SNPs and indels) between our sample and the reference genome. We have tried to mitigate a few errors along the way:
+
+1. Sequencing errors: these were addressed by the quality trimming step
+2. Sample contamination: we used `kraken2` to assess the extent of this problem in our sample
+3. Appropriate choice of a reference genome: we used a genome that is inferred to be ancestral to all *M. tuberculosis* for our analysis and the diversity within Mtb is limited enough for us to rely on a single reference genome for the entire species.
+4. Quality filtering in the mapping and variant calling stage: Tools like `bwa-mem` and `freebayes` judge the quality of their predictions, and `snippy` performs some filtering on variant calling predictions.
```

Comment:
It's not clear why you're refering to bwa-mem and freebayes, are they used by snippy?

Reply:
Yes, snippy does mapping with `bwa-mem` and then variant calling with `freebayes`.

Reply:
Thanks, I have tried to clarify the use of `bwa-mem` and `freebayes`.